.github/workflows/test.yaml

```
@@ -1,13 +1,16 @@
 ---
 name: test
-on:
-  push:
-    tags:
-      - v*
+on: pull_request
 jobs:
   test:
     runs-on: ubuntu-24.04
     steps:
-      - run: echo "$PAYLOAD"
+      - uses: actions/checkout@v4
```

action is not pinned
dquote> ```suggestion
      - uses: actions/checkout@v4.2.2
```